### PR TITLE
plugins/navic: migrate to mkNeovimPlugin

### PR DESCRIPTION
**Pull request by khaneliman**

### CODE REVIEW

File: plugins/bufferlines/navic.nix [hunk 1]
```diff
@@ -45,12 +66,12 @@ in
     };
 
     lsp = {
-      autoAttach = helpers.defaultNullOpts.mkBool false ''
+      autoAttach = defaultNullOpts.mkBool false ''
```
Comment (MattSturgeon) [inline plugins/bufferlines/navic.nix:69]: It seems option names haven't been converted to snake_case. [Checking upstream](https://github.com/SmiteshP/nvim-navic?tab=readme-ov-file#-customise), it seems they should be.

**Review by MattSturgeon — CHANGES_REQUESTED** [review]

File: plugins/bufferlines/navic.nix (hunk 1 above)
Comment (khaneliman) [reply plugins/bufferlines/navic.nix:69]: i think I accidentally left it while testing the deprecations..

**Review by MattSturgeon — APPROVED** [review]
:heart:

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**